controllers/designateworker_controller.go

```diff
@@ -798,12 +798,28 @@ func (r *DesignateWorkerReconciler) generateServiceConfigMaps(
 // returns the hash, whether the hash changed (as a bool) and any error
 func (r *DesignateWorkerReconciler) createHashOfInputHashes(
 	ctx context.Context,
+	h *helper.Helper,
 	instance *designatev1beta1.DesignateWorker,
 	envVars map[string]env.Setter,
 ) (string, bool, error) {
 	Log := r.GetLogger(ctx)
 	var hashMap map[string]string
 	changed := false
+
+	// If DesignateBindKeySecret exists, add its hash to status hash
+	rndcSecret := &corev1.Secret{}
+	err := h.GetClient().Get(ctx, types.NamespacedName{
+		Name:      designate.DesignateBindKeySecret,
+		Namespace: instance.Namespace,
+	}, rndcSecret)
+	if err == nil {
+		secretHash, err := secret.Hash(rndcSecret)
+		if err != nil {
+			return secretHash, changed, err
+		}
+		envVars[designate.DesignateBindKeySecret] = env.SetValue(secretHash)
+	}
```

Comment:
Shouldn't this return the err if it's not nil?

Reply:
Yes it should, I will add it.